src/main/webapp/app/exam/participate/exam-participation.component.ts

```diff
@@ -90,35 +97,17 @@ export class ExamParticipationComponent implements OnInit, OnDestroy {
             // init studentExam and activeExercise
             this.studentExam = studentExam;
             this.activeExercise = studentExam.exercises[0];
+            if (this.activeExercise.studentParticipations && this.activeExercise.studentParticipations.length === 0) {
+                // subscribe to execute
+                this.createParticipationForExercise(this.activeExercise).subscribe();
```

Comment:
There is another edge case. It could be that starting the exercise did not work properly and studentParticipations[0].initializationState is not yet set to INITIALIZED, but to any "previous" enum value.

Then, we also need to call startExercise on the server. The server will automatically determine the right steps to make sure the participation initialization is finished properly.

This is also the case for the course dashboard. If a participation is not fully initialized, Artemis still shows the "start exercise" button. You can also check exercise-utils.ts, however it uses a rather complex calculation which should not be necessary here.

Reply:
added a check for that